web/app/source/nvwrapper.coffee

```diff
@@ -3,52 +3,166 @@ $ = require 'jquery'
+      if chart.useInteractiveGuideline?
+        chart.useInteractiveGuideline(true)
+      chart.xAxis.axisLabel('Time').tickFormat (d) -> d3.time.format('%H:%M') new Date(d)
+      chart.yAxis.axisLabel('Temp (C)').tickFormat d3.format('.1f')
+      nv.utils.windowResize(chart.update)
+      new ChartDrawer target, chart
+
+
+ChartDrawer = class ChartDrawer
+
+  constructor: (@target, @chart) ->
+    @oldlast = 0
+
+  #
+  # withinCapacity basically will take a number of rows,
+  # and trim it by the max capacity
+  #
+  withinCapacity: (n) -> n
+
+  __structure: (data) ->
+    for f in __TS_FIELDS
+      key: data.channel[f]
+      values:
+        (x: parseDateStr(d.created_at), y: parseFloat(d[f]) for d in data.feeds)
+
+  drawChart: (data) ->
+    newdata = @__structure(data)
+    field1 = newdata[1]
+    last = field1.values[field1.values.length-1]
+    if last isnt @oldlast # if updated
+      @oldlast = last
+      d3.select(@target).datum(newdata).call @chart
+
+
+BarChartDrawer = class BarChartDrawer extends ChartDrawer
+
+  #
+  # because the bar chart only wants 1 row, it doesn't
+  # care about parameters, and just returns 1
+  #
+  withinCapacity: -> 1
+
+  __structure: (data) ->
+    newData = [{ key: 'all', values: []}]
+    for attribute in super data
+      newData[0].values.push
+        label: attribute.key
+        value: attribute.values[0].y
+    return newData
+
+  drawChart: (data) ->
+    newdata = @__structure(data)
+    d3.select(@target).datum(newdata).call @chart
+
+
+EggChartDrawer = class EggChartDrawer
+
+  constructor: (target) ->
+    @svg = d3.select(target)
+
+  #
+  # because the bar chart only wants 1 row, it doesn't
+  # care about parameters, and just returns 1
+  #
+  withinCapacity: -> 1
+
+  #
+  # parse an amount of px's like `250px' to `250'
+  #
+  __parsePx: (s) ->
+    parseInt s.substr(0, s.length - 2), 10
+
+  __structure: (data) ->
+    for f in __TS_FIELDS
+      label: data.channel[f]
+      value: (parseFloat(d[f]) for d in data.feeds)[0]
+
+  drawChart: (data) ->
+    formatted = @__structure data
+
+    circle = @svg.selectAll('image').data(formatted)
+    text   = @svg.selectAll('text').data(formatted)
+
+    radius = @__parsePx(@svg.style('height')) / 4.75
+    width  = @__parsePx(@svg.style 'width') - radius*3
+
+    circle.enter().append('image')
+      .attr "y", (@__parsePx(@svg.style('height')) / 2) - (radius * 1.25)
+      .attr "x", (d, i) =>
+        (radius*1.5 + (i * (width / (formatted.length-1)))) - radius
+      .attr "width", radius * 2
+      .attr "height", radius * 2
+      .attr "xlink:href", (d) ->
+        #
+        # eggcellent
+        #
+        switch
+          when d.value > 75 then '/img/egg1.png'
+          when d.value > 72 then '/img/egg2.png'
+          when d.value > 68 then '/img/egg3.png'
+          when d.value > 64 then '/img/egg4.png'
+          when d.value > 60 then '/img/egg5.png'
+          else '/img/egg6.png'
```

Comment:
[kinda based off this too an extent](http://www.splendidtable.org/sites/default/files/Egg%20Chart%20USE.jpg)